### PR TITLE
map : shipyard attributes ; jobs : mining jobs and a few more large bulk delivery jobs in human space

### PR DESCRIPTION
**Pull request by EndrosG**
majors
+ infile documentation of currently used attributes, including some comments on their usage
+ shipyard attributes to all relevant human, coalition, hai, wanderer, as well as Kessel Sepret, Sapira Mereti, Sestor Ikfar - i.e. no shipyards so far for Kor Efreti, Quarg and Pug --- part of #2359

+ added various mining jobs, for human space only (so far) --- part of #2101
+ added large bulk and large rush deliveries to major shipyards, for human space only (so far)

minors
+ attributes hai to planet Darkwaste (did not have any attributes yet)
+ attributes pug to planets Pugglemug and Pugglequat (did not have any attributes yet, and did never receive any by events, should happen in "pug territory liberated")
+ korath attr. to "Korati Efreti"
+ urban attr. to Hai-home and Stonebreak
+ shipyard attributes in Geminus related events (war begins, geminus rebuilt)
- put moon / station as 2nd attribute after the region attribute
- reordered the blocks in the job file a little bit (moved large bulk and large rush deliveries upwards)
+ added two additional empty lines between the blocks in the job file (i.e. 3 empty lines now between blocks)
+ in hai space, I changed "manufacturing" to "factory" on 4 planets (Dustmaker, Heartvalley, Makerplace, and Stonebreak), because Icelake used "factory" and `hai jobs.txt` uses "factory", while "manufacturing" is never used.

notes
- checked all event files, actually checked all data files in master repository
- I did not change a few things which seemed odd to me:
  - in coalition space, it is "wealthy" instead of "rich"
  - planet Hope is not volcanic

### CODE REVIEW

File: data/jobs.txt [hunk 1]
```diff
@@ -1571,6 +1576,136 @@ mission "Cargo [4]"
...
+	job
+	repeat
+	to offer
+		random < 15
+	source
+		government Republic "Free Worlds" Syndicate Neutral
+	destination
+		distance 0 3
+		government Republic "Free Worlds" Syndicate Neutral
+		attributes syMajor
```
Comment (jafdy) [inline data/jobs.txt]: I think shipyard would be a better attribute.

File: data/jobs.txt [hunk 2]
```diff
@@ -1602,106 +1737,917 @@ mission "Bulk Delivery [1]"
...
+mission "Bulk Delivery [2]"
+	name "Bulk delivery to <planet>"
+	job
+	repeat
+	description "Deliver <cargo> to <destination>. Payment is <payment>."
+	cargo random 25 2 .05
+	to offer
+		random < 50
+	source
+		attributes mining textiles factory farming fishing oil
```
Comment (jafdy) [inline data/jobs.txt]: Is this line needed?

File: data/map.txt [hunk 3]
```diff
@@ -22450,7 +22475,7 @@ planet "Ablub's Invention"
 	"required reputation" 25
 
 planet Ada
-	attributes paradise mining
+	attributes paradise shipyard syMinor mining
```
Comment (jafdy) [inline data/map.txt]: Same here. Stating that the ship yard is "syMinor" seems unnecessary.

File: data/events.txt [hunk 4]
```diff
@@ -3333,6 +3334,7 @@ event "syndicate tech available"
 
 event "geminus rebuilt"
 	planet Geminus
+		attributes paradise shipyard syMajor factory mining
```
Comment (EndrosG) [inline data/events.txt]: the shipyard attribute did not exist as attribute until today either. I created all three (shipyard, syMinor, syMajor) and also added a documentation header to the map.txt file summarizing all currently used attributes.
syMajor describes "home of (xyz) shipyard", like home of the Navy shipyard.

File: data/map.txt (hunk 3 above)
Comment (EndrosG) [inline data/map.txt]: I intend to create jobs where you deliver ship components from some syMajor to some syMinor destination.
I.e. some parts are only manufactured in the big yards...

File: data/jobs.txt (hunk 2 above)
Comment (EndrosG) [inline data/jobs.txt]: I am not sure upon the original intention, and I did not touch the Bulk Delivery [2] mission. I only moved the entire Bulk delivery block.
But I assume that the attributes line there is intentional.
This way a factory world can offer more jobs than a vanilla world. Vanilla can offer Bulk [0]+[1], factory can offer [0]+[1]+[2] at the same time.

File: data/jobs.txt (hunk 1 above)
Comment (EndrosG) [inline data/jobs.txt]: It was my intention to deliver tungsten only to major shipyards. I know that this way the mission cant be offered in every place, which is intentional.

**Comment by EndrosG** [comment]
> I get why you added "syMajor" and "syMinor" but it feels pointless as any planet (with the possible exception of mining worlds) will need some of most kinds of metals.

Many of the "small mining" jobs (except tungsten and uranium) apply to all shipyards.
The larger quantities have other audience, i.e. they are only asked by places with higher demands such as major shipyards. I dont think that the shipyard on "New Boston" (there are many similar ones) has a demand for large quantities, or for uranium at all.

If you think, more mining stuff should be added, like more mining for factory planets in general, feel free to add more.
I am not sure, how many new jobs of that type should be added in one step as we currently have none of that type at all and while it will enrich gameplay, it is not my intention to totally shift it from carrying cargo to "organizing" cargo. That's why i decided for lower values in the random conditions too. Let's first introduce them with a certain variety and eventually later modify them in terms of frequency, variants, sizing (upscaling?).

**Comment by EndrosG** [comment]
@endless-sky : what I can do / improve to get this merged?
Or asking different: Is there any chance you will merge this *after* you changed the CONTRIBUTING.md document? (this PR was made before you edited the latter)

I know it's a certain number of lines, but content needs a few more lines than code if variety is wanted.

**Comment by endless-sky** [comment]
Closing because there's a more recent PR with these additions (#3076).

**Pull request closed without merging**